File: ai-rag-app/ai-rag-app/Models/OshaModels.swift
//
//  OshaModels.swift
//  ai-rag-app
//
//  Created by Graham Diehl on 8/26/26.
//

import Foundation

struct OshaStandardQuery: Codable {
    let query: String
    let results: Int
}

struct OshaStandardResponse: Codable, Equatable {
    let answer: String
    let sources: [OshaSource]
}

struct OshaSource: Codable, Equatable, Identifiable {
    let citation: String
    let label: String
    let sourceUrl: String
    let text: String

    var id: String { citation + label }

    enum CodingKeys: String, CodingKey {
        case citation
        case label
        case sourceUrl = "source_url"
        case text
    }
}

struct OshaValidationError: Codable {
    let detail: String
}

File: ai-rag-android/app/src/main/java/net/apperdashery/demo/ai_rag_android/model/OshaModels.kt
package net.apperdashery.demo.ai_rag_android.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OshaStandardQuery(
    val query: String,
    val results: Int = 5
)

@Serializable
data class OshaStandardResponse(
    val answer: String,
    val sources: List<OshaSource>
)

@Serializable
data class OshaSource(
    val citation: String,
    val label: String,
    @SerialName("source_url") val sourceUrl: String,
    val text: String
)

@Serializable
data class OshaValidationError(
    val detail: String
)
